cmd/prometheus/main.go

```diff
@@ -186,6 +187,8 @@ func main() {
 
 	a.Flag("query.max-concurrency", "Maximum number of queries executed concurrently.").
 		Default("20").IntVar(&cfg.queryConcurrency)
+	a.Flag("query.max-samples", "Maximum number of samples a single query can return. Queries will fail if they would return more than this number.").
```

Comment:
This message isn't right, it's about samples in memory rather than returned.

Could we express it in bytes? It's a lower bound due to things like labelsets and capacity vs len, but we could improve it over time.